ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineGetPayloadBodiesByRangeV1.java

```diff
@@ -64,10 +64,15 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext request) {
         () -> count);
 
     final Blockchain blockchain = protocolContext.getBlockchain();
+
+    final long latestKnownBlockNumber = blockchain.getChainHeadBlockNumber();
+    final long upperBound = startBlockNumber + count;
+    final long finalBlockNumber = latestKnownBlockNumber < upperBound ? latestKnownBlockNumber + 1 : upperBound;
```

Comment:
suggestion: s/finalBlockNumber/endExclusiveBlockNumber (to match LongStream.range's naming)

Reply:
Done